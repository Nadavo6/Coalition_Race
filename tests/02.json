{
    "parties": [
      {
        "name": "Arctic Monkeys",
        "mandates": 25,
        "join_policy": "L"
      },
      {
        "name": "Muse",
        "mandates": 25,
        "join_policy": "L"
      },
      {
        "name": "Twenty One Pilots",
        "mandates": 25,
        "join_policy": "L"
      },
      {
        "name": "Foo Fighters",
        "mandates": 45,
        "join_policy": "L"
      }
    ],
    "graph": [
      [0, 1, 0, 0],
      [1, 0, 1, 0],
      [0, 1, 0, 1],
      [0, 0, 1, 0]
    ],
    "agents": [
      {
        "party_id": 0,
        "selection_policy": "E"
      }
    ]
  }
  