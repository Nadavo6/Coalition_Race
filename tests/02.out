[
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 25,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "Arctic Monkeys",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "Muse",
                "state": "Waiting"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "Twenty One Pilots",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 45,
                "name": "Foo Fighters",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 25,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "Arctic Monkeys",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "Muse",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "Twenty One Pilots",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 45,
                "name": "Foo Fighters",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 25,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "Arctic Monkeys",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "Muse",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "Twenty One Pilots",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 45,
                "name": "Foo Fighters",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 25,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "Arctic Monkeys",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "Muse",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "Twenty One Pilots",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 45,
                "name": "Foo Fighters",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0,
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "Arctic Monkeys",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "Muse",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "Twenty One Pilots",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 45,
                "name": "Foo Fighters",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0,
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "Arctic Monkeys",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "Muse",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "Twenty One Pilots",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 45,
                "name": "Foo Fighters",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0,
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "Arctic Monkeys",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "Muse",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "Twenty One Pilots",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 45,
                "name": "Foo Fighters",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            },
            {
                "id": 2,
                "partyId": 2
            }
        ],
        "coalitions": [
            {
                "mandates": 75,
                "parties": [
                    0,
                    1,
                    2
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "Arctic Monkeys",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 25,
                "name": "Muse",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 25,
                "name": "Twenty One Pilots",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 45,
                "name": "Foo Fighters",
                "state": "CollectingOffers"
            }
        ]
    }
]
